cdap-gateway/src/main/java/io/cdap/cdap/gateway/router/handlers/AuthenticationHandler.java

```diff
@@ -219,6 +228,17 @@ public void onChange(ServiceDiscovered serviceDiscovered) {
     }
   }
 
+  /**
+   * Encrypt the given credential followed by a base64 encoding.
+   */
+  private String encryptUserCredential(String credential) throws CipherException {
+    if (credential == null ||
+      !sConf.getBoolean(Constants.Security.Authentication.USER_CREDENTIAL_ENCRYPTION_ENABLED, false)) {
+      return credential;
+    }
+    return new TinkCipher(sConf).encryptToBase64(credential.getBytes(StandardCharsets.UTF_8), null);
```

Comment:
may be move this `credential.getBytes(StandardCharsets.UTF_8)` to `encryptToBase64 ` so that it takes string and returns string? Otherwise callers will have to do the conversion.

Reply:
Done.

Reply:
Done. Added a new API encryptStringToBase64